core/src/test/java/org/apache/iceberg/TestSchemaUpdate.java

```diff
@@ -779,6 +779,22 @@ public void testAddRequiredColumnCaseInsensitive() {
         .hasMessage("Cannot add column, name already exists: ID");
   }
 
+  @Test
+  public void testAddMultipleRequiredColumnCaseInsensitive() {
+    Schema schema = new Schema(required(1, "id", Types.IntegerType.get()));
+
+    assertThatThrownBy(
+            () ->
+                new SchemaUpdate(schema, 1)
+                    .caseSensitive(false)
+                    .allowIncompatibleChanges()
+                    .addRequiredColumn("data", Types.StringType.get())
+                    .addRequiredColumn("DATA", Types.StringType.get())
+                    .apply())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot build lower case index: data and DATA collide");
```

Comment:
I'm good with this but It'd be nice if we can detect this in a different way and fail with a better error message that dosen't really mention anything about the internal indexing we do.  since it's really no worse then the case-sensitive case (and it's consistent) where if someone does ```add("data"....).add("data"...)``` it fails with the same message, I'm good with it.